cudamapper/tests/Test_CudamapperMatcher.cpp

```diff
@@ -17,11 +17,379 @@
+        void read_id_and_representation_to_all_its_sketch_elements(const std::vector<std::map<representation_t, ArrayBlock>>& val) { read_id_and_representation_to_all_its_sketch_elements_ = val; }
+        void representation_to_all_its_sketch_elements(const std::map<representation_t, ArrayBlock>& val) { representation_to_all_its_sketch_elements_ = val; }
+    private:
+        std::vector<position_in_read_t> positions_in_reads_;
+        std::vector<read_id_t> read_ids_;
+        std::vector<SketchElement::DirectionOfRepresentation> directions_of_reads_;
+        std::uint64_t number_of_reads_;
+        std::vector<std::string> read_id_to_read_name_;
+        std::vector<std::map<representation_t, ArrayBlock>> read_id_and_representation_to_all_its_sketch_elements_;
+        std::map<representation_t, ArrayBlock> representation_to_all_its_sketch_elements_;
+    };
+
+    TEST(TestCudamapperMatcher, CustomIndexTwoReads) {
+        // Two reads, all minimizers have the same representation
+        TestIndex test_index;
+
+        std::vector<position_in_read_t> positions_in_reads(100);
+        std::iota(std::begin(positions_in_reads), std::end(positions_in_reads), 0);
+        test_index.positions_in_reads(positions_in_reads);
+
+        std::vector<read_id_t> read_ids(100);
+        std::fill(std::begin(read_ids), std::next(std::begin(read_ids), 50), 0);
+        std::fill(std::next(std::begin(read_ids), 50), std::end(read_ids), 1);
+        test_index.read_ids(read_ids);
+
+        // no need for directions yet
+
+        test_index.number_of_reads(2);
+
+        // no need for read_id_to_read_name
+
+        std::vector<std::map<representation_t, ArrayBlock>> read_id_and_representation_to_all_its_sketch_elements(2);
+        read_id_and_representation_to_all_its_sketch_elements[0].emplace(0x023, ArrayBlock{0,50});
+        read_id_and_representation_to_all_its_sketch_elements[1].emplace(0x023, ArrayBlock{50,50});
+        test_index.read_id_and_representation_to_all_its_sketch_elements(read_id_and_representation_to_all_its_sketch_elements);
+
+        std::map<representation_t, ArrayBlock> representation_to_all_its_sketch_elements;
+        representation_to_all_its_sketch_elements.emplace(0x023, ArrayBlock{0,100});
+        test_index.representation_to_all_its_sketch_elements(representation_to_all_its_sketch_elements);
+
+        Matcher matcher(test_index);
+
+        const std::vector<Matcher::Anchor>& anchors = matcher.anchors();
+        ASSERT_EQ(anchors.size(), 2500);
+
+        for (std::size_t read_0_sketch_element = 0; read_0_sketch_element < 50; ++read_0_sketch_element) {
+            for (std::size_t read_1_sketch_element = 0; read_1_sketch_element < 50; ++read_1_sketch_element) {
+                ASSERT_EQ(anchors[read_0_sketch_element*50 + read_1_sketch_element].query_read_id_, 0) << read_0_sketch_element << " " <<read_1_sketch_element;
```

Comment:
Should we be using `EXPECT_EQ` in these situations so that if there is a failure all information is printed (subsequent checks in this function) ?

Reply:
That was my initial implementation, but it's highly likely that if one test fails all tests will fail and we'd end up with 50x50x4 error messages (or even more in the lower test).
If all checks in both big tests would fail that would cause ~400000 error messages.
An alternative would be to have first three checks do `EXPECT` and the last one `ASSERT`, assuming all four anchors would be wrong at the same time (which does not have to be the case). What would you suggest?

Reply:
I see your point. regarding 10 thousand error messages.. OK Let's leave it as is.